src/Features/Core/Portable/EditAndContinue/EditAndContinueWorkspaceService.cs

```diff
@@ -85,11 +85,38 @@ public async ValueTask StartDebuggingSessionAsync(Solution solution, IManagedEdi
                 captureMatchingDocuments ? await CommittedSolution.GetMatchingDocumentsAsync(solution, _compilationOutputsProvider, cancellationToken).ConfigureAwait(false) :
                 SpecializedCollections.EmptyEnumerable<KeyValuePair<DocumentId, CommittedSolution.DocumentState>>();
 
-            var newSession = new DebuggingSession(solution, debuggerService, _compilationOutputsProvider, initialDocumentStates, _debuggingSessionTelemetry, _editSessionTelemetry);
+            var runtimeCapabilities = await debuggerService.GetCapabilitiesAsync(cancellationToken).ConfigureAwait(false);
+            var capabilities = ParseCapabilities(runtimeCapabilities);
+
+            var newSession = new DebuggingSession(solution, debuggerService, capabilities, _compilationOutputsProvider, initialDocumentStates, _debuggingSessionTelemetry, _editSessionTelemetry);
             var previousSession = Interlocked.CompareExchange(ref _debuggingSession, newSession, null);
             Contract.ThrowIfFalse(previousSession == null, "New debugging session can't be started until the existing one has ended.");
         }
 
+        // internal for testing
+        internal static ManagedEditAndContinueCapability ParseCapabilities(ImmutableArray<string> capabilities)
+        {
+            var caps = ManagedEditAndContinueCapability.None;
+
+            foreach (var capability in capabilities)
+            {
+                caps |= capability switch
+                {
+                    "Baseline" => ManagedEditAndContinueCapability.Baseline,
+                    "AddMethodToExistingType" => ManagedEditAndContinueCapability.AddMethodToExistingType,
+                    "AddStaticFieldToExistingType" => ManagedEditAndContinueCapability.AddStaticFieldToExistingType,
+                    "AddInstanceFieldToExistingType" => ManagedEditAndContinueCapability.AddInstanceFieldToExistingType,
+                    "NewTypeDefinition" => ManagedEditAndContinueCapability.NewTypeDefinition,
+
+                    // To make it eaiser for  runtimes to specify more broad capabilities
+                    "AddDefinitionToExistingType" => ManagedEditAndContinueCapability.AddMethodToExistingType | ManagedEditAndContinueCapability.AddStaticFieldToExistingType | ManagedEditAndContinueCapability.AddInstanceFieldToExistingType,
```

Comment:
Not sure if this is being too clever, but also I imagine it would be a pain to try to get Mono and .NET runtimes to keep pace with the same breakdown of capabilities.

This could go further too, for example `AddFieldToExistingType = AddStaticFieldToExistingType | AddInstanceFieldToExistingType`, or breaking up `AddMethodToExistingType`. Not sure what the sweet spot is now, or if we just revisit later?